crates/biome_html_analyze/src/lint/nursery/use_vue_v_for_key.rs

```diff
@@ -0,0 +1,118 @@
+use biome_analyze::{
+    Ast, Rule, RuleDiagnostic, RuleDomain, RuleSource, context::RuleContext, declare_lint_rule,
+};
+use biome_console::markup;
+use biome_html_syntax::{AnyVueDirective, HtmlAttributeList, VueDirectiveArgument};
+use biome_rowan::{AstNode, AstNodeList, TextRange};
+use biome_rule_options::use_vue_v_for_key::UseVueVForKeyOptions;
+
+declare_lint_rule! {
+    /// Enforce that elements using `v-for` also specify a unique `key`.
+    ///
+    /// When rendering lists with `v-for`, Vue relies on a `key` to track elements efficiently.
+    /// The `key` can be provided via longhand `v-bind:key` or shorthand `:key`.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```vue,expect_diagnostic
+    /// <li v-for="item in items">{{ item }}</li>
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```vue
+    /// <li v-for="item in items" :key="item.id">{{ item }}</li>
+    /// ```
+    ///
+    /// ```vue
+    /// <li v-for="item in items" v-bind:key="item.id">{{ item }}</li>
+    /// ```
+    ///
+    pub UseVueVForKey {
+        version: "next",
+        name: "useVueVForKey",
+        language: "html",
+        recommended: true,
+        domains: &[RuleDomain::Vue],
+        sources: &[RuleSource::EslintVueJs("require-v-for-key").same()],
+    }
+}
+
+impl Rule for UseVueVForKey {
+    type Query = Ast<HtmlAttributeList>;
+    type State = TextRange;
+    type Signals = Option<Self::State>;
+    type Options = UseVueVForKeyOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Option<Self::State> {
+        let attrs = ctx.query();
+        let mut has_v_for = None;
+        let mut has_key = false;
+        for attr in attrs.iter() {
+            if let Some(dir_any) = attr.as_any_vue_directive() {
+                match dir_any {
+                    AnyVueDirective::VueDirective(dir) => {
+                        if dir.name_token().is_ok_and(|t| t.text_trimmed() == "v-for") {
+                            has_v_for = Some(dir.range());
+                        }
+                        if dir.name_token().is_ok_and(|t| t.text_trimmed() == "v-bind")
+                            && let Some(arg) = dir.arg()
+                            && is_v_bind_key_argument(&arg)
+                        {
+                            has_key = true;
+                        }
+                    }
+                    AnyVueDirective::VueVBindShorthandDirective(sh) => {
+                        if let Ok(arg) = sh.arg()
+                            && is_v_bind_key_argument(&arg)
+                        {
+                            has_key = true;
+                        }
+                    }
+                    _ => {}
+                }
```

Comment:
There's a lot of branching, which can be reduced with `.ok()?`

Reply:
Not in this case I don't think because the try operator will result in the function returning `None` if it fails, and not skipping the current iteration.